### ISSUE TITLE
Firefox screen-sharing & webcam streaming broken on Ubuntu

### ISSUE THREAD

**Issue by shiftee** [jitsi/jitsi-meet#2350]
When one side is using Firefox then the other side doesn't seem to receive any screen-sharing data or webcam stream.

I sometimes get the message "Video for Fellow Jitster has been turned off to save bandwidth"

On the local side I can see the webcam-stream or shared window.
On the remote side it says "Member has stopped the camera" and the screen is black or shows a person icon.

Stopping the camera on one side has no visible effect on the other.

Using Firefox 57.0.3 on Ubuntu 16.04.

In summary:
Firefox can see screen-sharing or webcam-stream from chromium.
Chromium/Firefox cannot see screen-sharing or webcam-stream from Firefox.

It seems like firefox is not transmitting the stream or perhaps is doing so too aggressively and is being throttled

**Comment by shiftee** [comment]
Here's the contents of the browser console when doing a full refresh:

[browser_console.log](https://github.com/jitsi/jitsi-meet/files/1603287/browser_console.log)

**Comment by shiftee** [comment]
Camera streaming seems to be working now, although sometimes it takes 20-30 seconds to start.

Screen-sharing still fails. The remote side's screen goes blank.
If I stop screen-sharing video does not resume on the remote side and the local side's video is frozen.

Refreshing the webpage on remote computer does not fix issue.

Refreshing local webpage allows us to see video from remote side but remote side does not see video from local session.

Refreshing remote side again now restores two-way video streaming

**Comment by tuxayo** [comment]
Which versions? (Jitsi, Browser) To know what could have caused this progress.

**Comment by shiftee** [comment]
Firefox 57.0.4 (64-bit).

I'm testing with meet.jit.si so I guess it should be the latest release

**Comment by sgala** [comment]
Firefox 58

jitsi:
```
# dpkg -l *jitsi*
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                        Version            Architecture       Description
+++-===========================-==================-==================-===========================================================
ii  jitsi-meet                  1.0.2784-1         all                WebRTC JavaScript video conferences
ii  jitsi-meet-prosody          1.0.2570-1         all                Prosody configuration for Jitsi Meet
ii  jitsi-meet-web              1.0.2570-1         all                WebRTC JavaScript video conferences
ii  jitsi-meet-web-config       1.0.2570-1         all                Configuration for web serving of Jitsi Meet
ii  jitsi-videobridge           1029-1             amd64              WebRTC compatible Selective Forwarding Unit (SFU)


```
Video works well, screen sharing gives an error about the extension, but in the console on can see a huge traceback with the error:

```
TypeError
columnNumber: 417854
fileName: "https://example.org/libs/lib-jitsi-meet.min.js?v=2570"
lineNumber: 2
message: "e.split is not a function"
stack: "t@https://example.org/libs/lib-jitsi-meet.min.js?v=2570:2:417854\nr@https://example.org/libs/lib-jitsi-meet.min.js?v=2570:2:418042\nvalue@https://example.org/libs/lib-jitsi-meet.min.js?v=2570:2:411304\nvalue@https://example.org/libs/lib-jitsi-meet.min.js?v=2570:2:411402\nobtainScreenOnFirefox@https://example.org/libs/lib-jitsi-meet.min.js?v=2570:2:806244\nvalue/<@https://example.org/libs/lib-jitsi-meet.min.js?v=2570:2:209864\nO@https://example.org/libs/app.bundle.min.js?v=2570:2:2059763\nvalue@https://example.org/libs/lib-jitsi-meet.min.js?v=2570:2:209838\nvalue/n<@https://example.org/libs/lib-jitsi-meet.min.js?v=2570:2:212699\nvalue@https://example.org/libs/lib-jitsi-meet.min.js?v=2570:2:214110\nvalue@https://example.org/libs/lib-jitsi-meet.min.js?v=2570:2:194872\ncreateLocalTracks@https://example.org/libs/lib-jitsi-meet.min.js?v=2570:2:358617\nt.a@https://example.org/libs/app.bundle.min.js?v=2570:2:713801\n_createDesktopTrack@https://example.org/libs/app.bundle.min.js?v=2570:2:2224676\n_switchToScreenSharing@https://example.org/libs/app.bundle.min.js?v=2570:2:2225680\ntoggleScreenSharing@https://example.org/libs/app.bundle.min.js?v=2570:2:2224378\nn.prototype.emit@https://example.org/libs/app.bundle.min.js?v=2570:2:1339486\nx.emitEvent@https://example.org/libs/app.bundle.min.js?v=2570:2:2319295\nonClick@https://example.org/libs/app.bundle.min.js?v=2570:2:411069\nonClick@https://example.org/libs/app.bundle.min.js?v=2570:2:431564\nvalue@https://example.org/libs/app.bundle.min.js?v=2570:2:422661\nvalue@https://example.org/libs/app.bundle.min.js?v=2570:2:1284542\nu@https://example.org/libs/app.bundle.min.js?v=2570:2:802545\ninvokeGuardedCallback@https://example.org/libs/app.bundle.min.js?v=2570:2:865260\ninvokeGuardedCallbackAndCatchFirstError@https://example.org/libs/app.bundle.min.js?v=2570:2:865354\np@https://example.org/libs/app.bundle.min.js?v=2570:2:803355\nm@https://example.org/libs/app.bundle.min.js?v=2570:2:803826\ng@https://example.org/libs/app.bundle.min.js?v=2570:2:803983\nv@https://example.org/libs/app.bundle.min.js?v=2570:2:803625\nk@https://example.org/libs/app.bundle.min.js?v=2570:2:804658\nAr@https://example.org/libs/app.bundle.min.js?v=2570:2:880188\nke@https://example.org/libs/app.bundle.min.js?v=2570:2:811595\nbatchedUpdates@https://example.org/libs/app.bundle.min.js?v=2570:2:847001\nJ@https://example.org/libs/app.bundle.min.js?v=2570:2:808151\nSe@https://example.org/libs/app.bundle.min.js?v=2570:2:812043\n"
```

(This is running in my own server, I edited the hostname)

**Comment by hjwp** [comment]
hi, am not a jitsi user, but I found this thread when googling for a similar issue, and eventually found my way to this conversation which suggests it's a bug/missing feature in Wayland (the experimental replacement for Xorg, which is used in ubuntu 17.10) https://superuser.com/questions/1221333/screensharing-under-wayland

looks like the next release of ubuntu will stop using wayland, which might fix this issue.

hope that helps!

**Comment by RossComputerGuy** [comment]
Has anyone tried Jitsi on Wayland with the XDG Desktop Portal running? If Jitsi uses WebRTC then that will fix the issue.

**Comment by phrxmd** [comment]
> Has anyone tried Jitsi on Wayland with the XDG Desktop Portal running? If Jitsi uses WebRTC then that will fix the issue.

Yes. It works fine, tested with Firefox and Chromium.
